core/templates/dev/head/player/PlayerServices.js

```diff
@@ -185,21 +197,28 @@ oppia.factory('oppiaPlayerService', [
       // Record the state hit to the event handler.
       var stateHitEventHandlerUrl = '/explorehandler/state_hit_event/' + _explorationId;
 
-      if (newStateName) {
-        $http.post(stateHitEventHandlerUrl, {
-          new_state_name: newStateName,
-          exploration_version: version,
-          session_id: sessionId,
-          client_time_spent_in_secs: stopwatch.getTimeInSecs(),
-          old_params: learnerParamsService.getAllParams()
-        });
-      }
+      $http.post(stateHitEventHandlerUrl, {
+        new_state_name: newStateName,
+        exploration_version: version,
+        session_id: sessionId,
+        client_time_spent_in_secs: stopwatch.getTimeInSecs(),
+        old_params: learnerParamsService.getAllParams()
+      });
+
+      // Broadcast the state transition to the parent page.
+      messengerService.sendMessage(messengerService.STATE_TRANSITION, {
+        oldStateName: _currentStateName,
+        jsonAnswer: JSON.stringify(answer),
+        newStateName: newStateName ? newStateName : 'END'
```

Comment:
Is this logic still necessary? The end state must exist, so 'newStateName' should always be defined. Plus, I'm fairly certain any logic that checked 'END' before should now check is_terminal on the interaction of that state.